### DISCUSSION TITLE
Complex Report question - Variable Field names in a report.

### DISCUSSION THREAD

**Discussion by EAwesomeK** [eclipse-birt/birt#1210] [Q&A]
I am fairly new to BIRT but I need to write a report that may be fairly complex.  I want to run a report which will have a variable number of fields based upon value of one piece of data and I am not sure how to approach it.  Here is an example:

Item1 - Number of records=42
Fields with data    Data1, data2, data3, ... data42

Items 2  Number of Records=30
Fields with data    Data1, data2, data3, ... data30

While the field names are the same and are numbered 1 to 42 do I have to add all of the fields to the report layout or can I have a for next loop of some sort add the fields to the data section but using variable such as DataX for 1 to number of records.
Any questions please dont hesitate to ask
Any help would be much appreciated.

**Comment by wimjongman** [comment]
It helps if you provide a mockup of the report you envision. The description you provide is subject to multiple interpretations.

**Reply by speckyspooky** [reply]
I confirm wimjongmans detailed request to support. Because dependend on your environment and details it would be possible to use special SQL-statements dependend on your source of data or may be you use flat files like data source there could be loops with dynamic arrays helpfull, so we have many options on report side.

**Comment by EAwesomeK** [comment]
![Birt complex report](https://user-images.githubusercontent.com/37028193/218187202-126b4201-71bb-47b6-81af-f9b8056a9828.PNG)

The circled area is the number of items in that particular instance, the number of breakers.  The larger area that is outlined is the actualy breakers, there are 42 fields for each column, breaker1, load1, amps1, breaker2, load2, amps 2, etc.

In another instance there may only be 24 breakers.  Do you understand?

**Reply by speckyspooky** [reply]
I miss the original source of your explanation. I mean what is your data base.
Because it seems to me that you work with "grouping" and if you would use a standard data set e.g. based on dataset (SQL) or scripted dataset you can display all your data group in the detail section. And in this case you haven't any problem with the number of data.

Your target must be that you get our different data arranged in the same data structure of a data set,
I mean:
  - if you data have in each row the same column count indepoendend of data filled your can used it.
  - you must have only the full count of columns of the maximum data count

But all my ideas want be help you because we didn't know your original data situation and we haven't a drafting version.
So it is very hard to assist you.

Perhaps it will help you. If you could provide use a small example we can go forward with clarification.
